promql/engine.go

```diff
@@ -1048,6 +1105,33 @@ func (ev *evaluator) eval(expr Expr) Value {
 			panic(fmt.Errorf("cannot do range evaluation of matrix selector"))
 		}
 		return ev.matrixSelector(e)
+
+	case *SubqueryExpr:
+		newEv := &evaluator{
+			endTimestamp: ev.endTimestamp - int64(e.Offset/time.Millisecond),
+			interval:     int64(getGlobalEvaluationInterval() / time.Millisecond),
+			ctx:          ev.ctx,
+			maxSamples:   ev.maxSamples,
+			logger:       ev.logger,
+		}
+		if e.StepExists {
+			newEv.interval = int64(e.Step / time.Millisecond)
+		}
+
+		// Calculate number of steps of the subquery evaluation.
+		rangeMillis := (int64(e.Range / time.Millisecond))
+		numSteps := (rangeMillis + (newEv.interval / 2)) / newEv.interval
```

Comment:
Why are you adding in half?

Reply:
What I understood was, the start time should be integral steps away from end time. And that it should either overlap with the (end-range), or should be the one just behind it if it doesn't exactly overlap. So adding that half takes case of just behind it.

But oh well, I just realised adding that half won't take care of it, I guess I mixed something here. I will fix it. But is my above explanation right?